### ISSUE TITLE
Search thread hangs

### ISSUE THREAD

**Issue by kaysond** [AHAAAAAAA/PokemonGo-Map#1107]
Deployed on Heroku. For some reason, the search thread just stops after a while. No errors in the log, just see:
2016-07-22T02:17:50.363162+00:00 app[web.1]: 2016-07-22 02:17:50,362 [     search] [   INFO] Skipping Pokemon Go login process since already logged in for another 1467.64 seconds

2016-07-22T02:17:50.363267+00:00 app[web.1]: 2016-07-22 02:17:50,363 [     search] [   INFO] Scanning step 1 of 25.

And then nothing else from the search thread. Restarting dynos solves the problem.

**Comment by Krys1s** [comment]
Not sure if it is the exact same thing but I am getting

`2016-07-21 19:36:11,414 [     search] [  ERROR] Scan step failed. Response dicti
onary key error.
Exception in thread search_thread:
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "C:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "C:\Pogomap\pogom\search.py", line 99, in search_loop
    search(args)
  File "C:\Pogomap\pogom\search.py", line 87, in search
    failed_consecutive += 1
UnboundLocalError: local variable 'failed_consecutive' referenced before assignm
ent`

Running multiple instances, over the course of about an hour most are errored out with that message

**Comment by niayh** [comment]
Also same thing here. Sometimes it happens instantly before the first scan even happens, sometimes at the end of the first loop and sometimes for a couple.
```
2016-07-21 17:01:15,491 [     search] [  ERROR] Scan step failed. Response dictionary key error.
Exception in thread search_thread:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/username/PokemonGo-Map-GPS/pogom/search.py", line 99, in search_loop
    search(args)
  File "/home/username/PokemonGo-Map-GPS/pogom/search.py", line 87, in search
    failed_consecutive += 1
UnboundLocalError: local variable 'failed_consecutive' referenced before assignment
```

**Comment by jakon2048** [comment]
in pogom/search.py, adding `global failed_consecutive` should fix the problem

`            log.error('Scan step failed. Response dictionary key error.')
            global failed_consecutive
            failed_consecutive += 1`

**Comment by Krys1s** [comment]
Ignorance here -- is there something needed after adding that? I add that to the file and then the script fails to launch altogether.

**Comment by niayh** [comment]
I'm no python guru, but the syntax of that statement was off. I pushed
```
global failed_consecutive
```
to the next line and am currently testing to see if it works

**Comment by jakon2048** [comment]
` 
except KeyError:
            log.error('Scan step failed. Response dictionary key error.')
            global failed_consecutive
            failed_consecutive += 1
`

it should look like the above

**Comment by kaysond** [comment]
I'm getting a new error now...

2016-07-22T03:57:54.384315+00:00 app[web.1]: Exception in thread search_thread:
2016-07-22T03:57:54.384324+00:00 app[web.1]: Traceback (most recent call last):
2016-07-22T03:57:54.384335+00:00 app[web.1]:     self.__target(*self.__args, **self.__kwargs)
2016-07-22T03:57:54.384335+00:00 app[web.1]:   File "/app/pogom/search.py", line 99, in search_loop
2016-07-22T03:57:54.384336+00:00 app[web.1]:     search(args)
2016-07-22T03:57:54.384338+00:00 app[web.1]: 
2016-07-22T03:57:54.384333+00:00 app[web.1]:     self.run()
2016-07-22T03:57:54.384315+00:00 app[web.1]: Exception in thread search_thread

Not sure if it's a heroku issue or an issue with the thread dying due to bad server responses or anything...

**Comment by Krys1s** [comment]
Got it -- having it on separate lines worked.
`except KeyError:
            log.error('Scan step failed. Response dictionary key error.')
            global failed_consecutive
            failed_consecutive += 1
            if(failed_consecutive >= 5):`

**Comment by kaysond** [comment]
Most people's issue (failed_consecutive) is fixed in this commit: dfe9b88a77c1480e76e4ab5fff7d9b074bf0b59c

My problem is separate so I'll open a new issue.
